### PR TITLE
Add files, scripts and actions for linting and static analysis

### LINKED ISSUE #2: Include static checks
- [x] mypy
- [x] bandit
- [x] linting

### PR DESCRIPTION
**Pull request by kumaranvpl**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,6 +11,71 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
-dev = [
+lint = [
+    "types-PyYAML",
+    "types-setuptools",
+    "types-ujson",
+    "mypy==1.5.1",
```
Comment (davorrunje) [inline pyproject.toml:24]: this is not linting, this is static analysis

File: pyproject.toml [hunk 2]
```diff
@@ -11,6 +11,71 @@ dependencies = [
...
+install_types = true
+non_interactive = true
+
+# from https://blog.wolt.com/engineering/2021/09/30/professional-grade-mypy-configuration/
+disallow_untyped_defs = true
+no_implicit_optional = true
+check_untyped_defs = true
+warn_return_any = true
+show_error_codes = true
+warn_unused_ignores = false
```
Comment (davorrunje) [inline pyproject.toml]: this should be true by default

File: pyproject.toml [hunk 3]
```diff
@@ -11,6 +11,71 @@ dependencies = [
...
+
+disallow_incomplete_defs = true
+disallow_untyped_decorators = true
+disallow_any_unimported = false
+
+[tool.isort]
```
Comment (davorrunje) [inline pyproject.toml]: we are not using isort anywhere

File: scripts/lint.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+echo "Running pyup_dirs..."
+pyup_dirs --py38-plus --recursive app tests
+
+echo "Running ruff..."
+ruff app tests --fix
+
+echo "Running isort..."
```
Comment (davorrunje) [inline scripts/lint.sh]: ruff will handle sorting of import, please remove this

File: scripts/start_kafka_broker_locally.sh
Comment (davorrunje) [inline scripts/start_kafka_broker_locally.sh:1]: why is this empty?

**Review by davorrunje — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (kumaranvpl) [reply pyproject.toml:24]: Fixed it here but please check faststream repo https://github.com/airtai/faststream/blob/main/pyproject.toml#L89

File: pyproject.toml (hunk 2 above)
Comment (kumaranvpl) [reply pyproject.toml]: Fixed it here but please check faststream repo https://github.com/airtai/faststream/blob/main/pyproject.toml#L153

File: pyproject.toml (hunk 3 above)
Comment (kumaranvpl) [reply pyproject.toml]: Fixed it here but please check faststream repo https://github.com/airtai/faststream/blob/main/pyproject.toml#L159

File: scripts/lint.sh (hunk 4 above)
Comment (kumaranvpl) [reply scripts/lint.sh]: Isort removed

File: scripts/start_kafka_broker_locally.sh
Comment (kumaranvpl) [reply scripts/start_kafka_broker_locally.sh:1]: Only file permission changed. I made them executable. So changes shows only the file permission change not the content change which is none.

**Review by davorrunje — APPROVED** [review]

**Pull request merged**